### PR TITLE
Allow overriding release name

### LINKED ISSUE #430: Allow overriding the hono release name
When deploying hono, it always prefixes the pods (amongst other things) with `.Release.Name`. Almost all other helm charts we use (including Ditto) allow overriding the release name with a custom value.

In our case, ArgoCD sets the Release name to something silly, so we'd like the option to specify the release name ourselves.   
See: https://github.com/eclipse/packages/blob/master/charts/ditto/templates/_helpers.tpl#L14 and https://github.com/eclipse/packages/blob/master/charts/ditto/templates/connectivity-deployment.yaml#L17

### PR DESCRIPTION
**Pull request by WatcherWhale**
This is a continuation of #416 and fixes #430

### CODE REVIEW

File: charts/hono/templates/_helpers.tpl [hunk 1]
```diff
@@ -16,7 +16,8 @@
 Expand the name of the chart.
```
Comment (calohmn) [inline charts/hono/templates/_helpers.tpl:16]: Would be good to add this here:
```
The scope passed in is expected to be a dict with keys
- (mandatory) "dot": the root (".") scope
```

Also applies to the documentation sections for:
- `define "hono.fullname"`
- `define "hono.chart"`
- `define "hono.std.labels`

File: charts/hono/templates/_helpers.tpl [hunk 2]
```diff
@@ -461,7 +464,7 @@ quarkus:
       exporter:
         otlp:
           {{- if .dot.Values.jaegerBackendExample.enabled }}
-          endpoint: {{ printf "http://%s-jaeger-collector:4317" .dot.Release.Name | quote }}
+          endpoint: {{ printf "http://%s-jaeger-collector:4317" ( include "hono.fullname" .dot ) | quote }}
```
Comment (calohmn) [inline charts/hono/templates/_helpers.tpl:467]: Installing the chart with `--set jaegerBackendExample.enabled=true` causes an error here:
```
Error: INSTALLATION FAILED: template: 
hono/templates/hono-service-device-registry-embedded/hono-service-device-registry-secret.yaml:78:8: 
executing "hono/templates/hono-service-device-registry-embedded/hono-service-device-registry-secret.yaml" 
at <include "hono.quarkusConfig" $args>: 
error calling include: template: hono/templates/_helpers.tpl:467:66: 
executing "hono.quarkusConfig" at <include "hono.fullname" .dot>: 
error calling include: template: hono/templates/_helpers.tpl:29:31: 
executing "hono.fullname" at <.dot.Values.fullnameOverride>: 
nil pointer evaluating interface {}.Values
```

To fix this, it should be 
`include "hono.fullname" .`
above instead of 
`include "hono.fullname" .dot`.

There are 3 more places in `_helpers.tpl` where `include "hono.fullname" .dot` needs to be replaced in the same way.

**Comment by calohmn** [comment]
@WatcherWhale There were changes committed in master (expired Hono Certs getting recreated). Can you rebase?

File: charts/hono/templates/_helpers.tpl (hunk 2 above)
Comment (WatcherWhale) [reply charts/hono/templates/_helpers.tpl]: done

File: charts/hono/templates/_helpers.tpl (hunk 1 above)
Comment (WatcherWhale) [reply charts/hono/templates/_helpers.tpl:16]: done

File: charts/hono/templates/_helpers.tpl [hunk 3]
```diff
@@ -263,7 +278,7 @@ The scope passed in is expected to be a dict with keys
 kafka:
 {{- if .dot.Values.kafkaMessagingClusterExample.enabled }}
   commonClientConfig:
-    {{- $bootstrapServers := printf "%[1]s-%[2]s-0.%[1]s-%[2]s-headless:%d" .dot.Release.Name .dot.Values.kafka.nameOverride ( .dot.Values.kafka.service.ports.client | int ) }}
+    {{- $bootstrapServers := printf "%[1]s-%[2]s-0.%[1]s-%[2]s-headless:%d" ( include "hono.fullname" . ) .dot.Values.kafka.nameOverride ( .dot.Values.kafka.service.ports.client | int ) }}
```
Comment (calohmn) [inline charts/hono/templates/_helpers.tpl:281]: In the `bootstrapServers` value, the `fullname` of the Kafka chart deployment has to be used.
This can be achieved by using this:
```
    {{- $kafkaNameValues := pick .dot.Values.kafka "nameOverride" "fullnameOverride" }}
    {{- $kafkaChartDotScope := dict "dot" (dict "Release" .dot.Release "Chart" (dict "Name" "kafka") "Values" $kafkaNameValues) }}
    {{- $bootstrapServers := printf "%[1]s-0.%[1]s-headless:%d" ( include "hono.fullname" $kafkaChartDotScope ) ( .dot.Values.kafka.service.ports.client | int ) }}
```
(Using `deepcopy` on the `.dot` dict caused the linting/deployment to get locked up in my tests.)

File: charts/hono/templates/_helpers.tpl [hunk 4]
```diff
@@ -482,10 +497,10 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- .Values.prometheus.server.fullnameOverride | trunc 63 | trimSuffix "-" -}}
 {{- else -}}
 {{- $name := default "prometheus" .Values.prometheus.nameOverride -}}
-{{- if contains $name .Release.Name -}}
-{{- printf "%s-%s" .Release.Name .Values.prometheus.server.name | trunc 63 | trimSuffix "-" -}}
+{{- if contains $name ( include "hono.fullname" . ) -}}
+{{- printf "%s-%s" ( include "hono.fullname" . ) .Values.prometheus.server.name | trunc 63 | trimSuffix "-" -}}
 {{- else -}}
-{{- printf "%s-%s-%s" .Release.Name $name .Values.prometheus.server.name | trunc 63 | trimSuffix "-" -}}
+{{- printf "%s-%s-%s" ( include "hono.fullname" . ) $name .Values.prometheus.server.name | trunc 63 | trimSuffix "-" -}}
```
Comment (calohmn) [inline charts/hono/templates/_helpers.tpl:503]: The Prometheus service resource is using a name created in the context of the Prometheus chart.
Therefore, using the Hono fullname isn't correct here.
The changes in these lines should just be reverted, using `.Release.Name` again instead of `( include "hono.fullname" . )`.

File: packages/cloud2edge/Chart.yaml [hunk 5]
```diff
@@ -11,7 +11,7 @@
 # SPDX-License-Identifier: EPL-2.0
 #
 apiVersion: v1
-version: 0.4.0
+version: 0.4.1
 appVersion: 0.4.0
```
Comment (calohmn) [inline packages/cloud2edge/Chart.yaml:15]: `appVersion` should be adapted as well.

File: charts/hono/templates/prometheus/prometheus-config.yaml [hunk 6]
```diff
@@ -14,7 +14,7 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  {{- $args := dict "dot" . "component" "metrics" "name" "prometheus-config" }}
+  {{- $args := dict "dot" (. | deepCopy | merge (dict "Values" (pick .Values.prometheus.server "nameOverride" "fullnameOverride"))) "component" "metrics" "name" "prometheus-config" }}
```
Comment (calohmn) [inline charts/hono/templates/prometheus/prometheus-config.yaml:17]: The Prometheus chart expects the ConfigMap name to be `{ReleaseName}-{.Values.prometheus.server.configMapOverrideName}`, i.e. `{ReleaseName}-prometheus-config`.
But the code above doesn't produce that with the current values.yaml.

I would suggest using the following:
```
  {{- /* prometheus chart expects metadata.name to be "{ReleaseName}-{prometheus.server.configMapOverrideName}" */}}
  {{- $prometheusDotScope := . | deepCopy | merge (dict "Values" (dict "fullnameOverride" .Release.Name)) }}
  {{- $args := dict "dot" $prometheusDotScope "component" "metrics" "name" .Values.prometheus.server.configMapOverrideName }}
```

File: charts/hono/values.yaml [hunk 7]
```diff
@@ -1708,7 +1711,7 @@ kafka:
       type: "pem"
       pemChainIncluded: true
       existingSecrets:
-      - "{{ .Release.Name }}-kafka-example-keys"
+      - "{{ include \"hono.fullname\" (dict \"dot\" .) }}-kafka-example-keys"
```
Comment (calohmn) [inline charts/hono/values.yaml:1714]: As the above value is evaluated in the context of the Kafka chart, the `hono.fullname` template will produce `{ReleaseName}-kafka` here, meaning the complete value will be `{ReleaseName}-kafka-kafka-example-keys`.
Therefore, the additional `-kafka' part can be omitted above.

We could also add a comment there, to document the special logic:
```
      - "{{/* evaluated in kafka chart via tpl; kafka chart full name used here! */}}\
      {{ include \"hono.fullname\" (dict \"dot\" .) }}-example-keys"
```
(see also corresponding comment regarding `kafka-example-keys.yaml`)

File: charts/hono/templates/kafka/kafka-example-keys.yaml [hunk 8]
```diff
@@ -14,7 +14,7 @@
 apiVersion: v1
 kind: Secret
 metadata:
-  {{- $args := dict "dot" . "component" "kafka" "name" "kafka-example-keys" }}
+  {{- $args := dict "dot" (. | deepCopy | merge (dict "Values" (pick .Values.kafka "nameOverride" "fullnameOverride"))) "component" "kafka" "name" "kafka-example-keys" }}
```
Comment (calohmn) [inline charts/hono/templates/kafka/kafka-example-keys.yaml:17]: Corresponding to the other comment about omitting the duplicate `-kafka` part, this could be changed to:
```
  {{- /* metadata.name is set to "{fullNameOfKafkaSubChart}-example-keys" here, meaning e.g. "{ReleaseName}-kafka-example-keys", not "{ReleaseName}-hono-example-keys" */}}
  {{- $args := dict "dot" (. | deepCopy | merge (dict "Values" (pick .Values.kafka "nameOverride" "fullnameOverride"))) "component" "kafka" "name" "example-keys" }}
```

File: charts/hono/README.md [hunk 9]
```diff
@@ -102,6 +102,10 @@ helm uninstall eclipse-hono -n hono
 The command removes all the Kubernetes components associated with the chart and deletes the release.
 
 ## Release Notes
+### 2.5.0
+
+* Allow customizing the pod/service names irrespective or .Release.Name
```
Comment (calohmn) [inline charts/hono/README.md:107]: "irrespective of .Release.Name."

File: packages/cloud2edge/Chart.yaml [hunk 10]
```diff
@@ -11,8 +11,8 @@
 # SPDX-License-Identifier: EPL-2.0
 #
 apiVersion: v1
-version: 0.4.1
-appVersion: 0.4.1
+version: 0.4.2
+appVersion: 0.4.2
```
Comment (calohmn) [inline packages/cloud2edge/Chart.yaml:15]: @WatcherWhale Could you revert the changes in the cloud2edge Chart.yaml? They are not needed.
(Changes to the Hono version requirement have already been included in PR #483.)

File: charts/hono/templates/dispatch-router/dispatch-router-secret.yaml [hunk 11]
```diff
@@ -20,6 +20,7 @@ metadata:
 type: Opaque
 stringData:
   "qdrouterd.json": |
+    {{- $_ := set $ "dot" $args.dot }}
```
Comment (sophokles73) [inline charts/hono/templates/dispatch-router/dispatch-router-secret.yaml:23]: is there a reason why `$` is not being used anywhere?

File: charts/hono/templates/hono-service-command-router/hono-service-command-router-deployment.yaml [hunk 12]
```diff
@@ -72,4 +72,4 @@ spec:
         {{- include "hono.component.healthChecks" $args | nindent 8 }}
       volumes:
       {{- include "hono.pod.volumes" $args | indent 6 }}
-      serviceAccountName: {{ printf "%s-%s" .Release.Name $args.name | quote }}
+      serviceAccountName: {{ printf "%s-%s" ( include "hono.fullname" . ) $args.name | quote }}
```
Comment (sophokles73) [inline charts/hono/templates/hono-service-command-router/hono-service-command-router-deployment.yaml:75]: don't we need to pass in `$args` here instead of `.`

File: charts/hono/example/add_example_data_device_registry.sh [hunk 13]
```diff
@@ -11,7 +11,7 @@
 #
 # SPDX-License-Identifier: EPL-2.0
 #*******************************************************************************
-HTTP_BASE_URL="http://{{ .Release.Name }}-service-device-registry:8080/v1"
+HTTP_BASE_URL="http://{{ include "hono.fullname" . }}-service-device-registry:8080/v1"
```
Comment (sophokles73) [inline charts/hono/example/add_example_data_device_registry.sh:14]: does this actually produce the desired outcome? Don't we need to use a `printf` statement here as well?

File: charts/hono/values.yaml [hunk 14]
```diff
@@ -1708,7 +1711,8 @@ kafka:
       type: "pem"
       pemChainIncluded: true
       existingSecrets:
-      - "{{ .Release.Name }}-kafka-example-keys"
+      # Evaluated in kafka chart via tpl; kafka chart full name used here!
+      - "{{ include \"hono.fullname\" (dict \"dot\" .) }}-example-keys"
```
Comment (sophokles73) [inline charts/hono/values.yaml:1715]: if invoking the template works here, couldn't we also use something like this in order to prevent having to escape the quotes?
```yaml
- {{- printf "%s-example-keys" ( include "hono.fullname" (dict "dot" . ) ) | quote }}
```

**Review by sophokles73 — COMMENTED** [review]
just some smaller issues
In general, can you also please adapt the copyright headers of the files you changed to reflect the current year?

Thanks for contributing :+1:

File: charts/hono/templates/_helpers.tpl [hunk 15]
```diff
@@ -14,34 +14,46 @@
...
-    {{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+  {{- $fullnameOverride := .dot.Values.fullnameOverride -}}
+  {{- if $fullnameOverride  -}}
+    {{- $fullnameOverride | trunc 63 | trimSuffix "-" -}}
   {{- else -}}
-    {{- $name := default .Chart.Name .Values.nameOverride -}}
-    {{- if contains $name .Release.Name -}}
-      {{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+    {{- $nameOverride := .dot.Values.nameOverride -}}
+    {{- $name := empty $nameOverride | ternary .dot.Chart.Name $nameOverride -}}
```
[the remark this answers is not in the document]
Comment (WatcherWhale) [reply charts/hono/templates/_helpers.tpl:42]: Not exactly, as `hono.name` is truncated to 63 chars and here we still need the full name to check if the release name matches

**Comment by calohmn** [comment]
@WatcherWhale There has been [an error](https://github.com/eclipse/packages/actions/runs/5974982602/job/16211198131?pr=469) running the checks for this PR:
```
 >>> helm lint charts/hono --values charts/hono/ci/mongodb-based-device-registry-values.yaml
==> Linting charts/hono
Error:  templates/: template: hono/charts/mongodb/templates/standalone/svc.yaml:12:14: 
 executing "hono/charts/mongodb/templates/standalone/svc.yaml" 
at <include "common.labels.standard" (dict "customLabels" .Values.commonLabels "context" $)>: 
error calling include: template: hono/charts/kafka/charts/common/templates/_labels.tpl:6:27: 
 executing "common.labels.standard" at <include "common.names.name" .>: 
error calling include: template: hono/charts/kafka/charts/common/templates/_names.tpl:6:18: 
 executing "common.names.name" at <.Chart.Name>: nil pointer evaluating interface {}.Name
```

This error can be reproduced locally like this:
```
cd charts/hono
helm dependency update
helm lint . --values ci/mongodb-based-device-registry-values.yaml
```
The dependency update will update the used MongoDB chart from `13.16.4` to `13.17.2`.  The new MongoDB chart version makes use of an updated template function in the bitnami `common` dependency version `2.9.0`.
But the referenced Kafka chart is still using an older `common` version which isn't compatible here, and that older `common` chart version seems indeed to be loaded. Hence the error above.

See https://github.com/bitnami/charts/issues/18890#issuecomment-1694733376.

The straightforward solution seems to be to prevent the usage of the MongoDB `13.17.x` chart version by using
```
- name: "mongodb"
  repository: "https://charts.bitnami.com/bitnami"
  version: "~13.16.x"
```
in the Hono `Chart.yaml`.

The alternative would be to update the Kafka chart to `25.0.0` (released last week). This version is also using `common` `2.9.0`.
But I think we should leave this Kafka chart version jump to a separate PR.

File: charts/hono/templates/hono-service-command-router/hono-service-command-router-secret.yaml
Comment (calohmn) [inline charts/hono/templates/hono-service-command-router/hono-service-command-router-secret.yaml:1]: The line 
`{{- tpl ( .Files.Get "config/command-router/embedded-cache-config.xml" ) . | nindent 4 }}` 
in this file needs to be replaced with
```
  {{- $_ := set $ "dot" $args.dot }}
  {{- tpl ( .Files.Get "config/command-router/embedded-cache-config.xml" ) $ | nindent 4 }}
```
.

File: charts/hono/templates/NOTES.txt [hunk 16]
```diff
@@ -14,41 +14,41 @@ The output should look similar to this:
 
 NAME                                       {{ "READY   STATUS    RESTARTS   AGE" | indent ( len .Release.Name ) }}
 {{- if .Values.adapters.amqp.enabled }}
-{{ .Release.Name }}-adapter-amqp-65cfb4d675-g5wn4             1/1     Running   0          5m50s
+{{ include "hono.fullname" . }}-adapter-amqp-65cfb4d675-g5wn4             1/1     Running   0          5m50s
```
[the remark this answers is not in the document]
Comment (calohmn) [reply charts/hono/templates/NOTES.txt:17]: Actually, I think we could get rid of this whole rather obscure `{{- $_ := set $ "dot" $args.dot }}` usage, by reverting the `define "hono.fullname"` named template changes - not requiring a `dot` dict there anymore but letting "hono.fullname" use `.Values` directly, instead of `.dot.Values`.

That would mean:
- replacing `include "hono.fullname" $args` with `include "hono.fullname" .` in the yaml files
- reverting the above `$_ := set $ "dot"` changes
- replacing `include "hono.fullname" .` with `include "hono.fullname" .dot` in `_helpers.tpl`
- adapting the place where `$kafkaChartDotScope` is defined within `_helpers.tpl`, removing the  outer `dict "dot"`
- replacing `include "hono.fullname" (dict "dot" .)` with `include "hono.fullname" .` in `values.yaml`
- adapting `kafka-example-keys.yaml`, using this in the `metadata` block:
```
  {{/* metadata.name is set to "{fullNameOfKafkaSubChart}-example-keys" here, meaning e.g. "{ReleaseName}-kafka-example-keys", not "{ReleaseName}-hono-example-keys" */}}
  {{- $kafkaNameValues := pick .Values.kafka "nameOverride" "fullnameOverride" }}
  {{- $kafkaChartDotScope := dict "Release" .Release "Chart" (dict "Name" "kafka") "Values" $kafkaNameValues }}
  {{- $args := dict "dot" ($kafkaChartDotScope) "component" "kafka" "name" "example-keys" }}
  {{- include "hono.metadata" $args | nindent 2 }}
```

@WatcherWhale @sophokles73 WDYT?

[the remark this answers is not in the document]
Comment (WatcherWhale) [reply charts/hono/templates/NOTES.txt:17]: It is indeed a rather obscure line, I'm in favor of changing it.

File: charts/hono/Chart.yaml [hunk 17]
```diff
@@ -46,7 +46,7 @@ dependencies:
   condition: "grafana.enabled"
 - name: "mongodb"
   repository: "https://charts.bitnami.com/bitnami"
-  version: "^13.x"
+  version: "~13.16.x"
```
Comment (sophokles73) [inline charts/hono/Chart.yaml:49]: is this related to the issue of overriding the release name?
If not, I would prefer to do any necessary changes to the Mongo DB version in a separate issue/PR ...

Comment (WatcherWhale) [reply charts/hono/Chart.yaml:49]: It is to fix this issue https://github.com/eclipse/packages/pull/469#issuecomment-1696238963

Comment (calohmn) [reply charts/hono/Chart.yaml:49]: Created separate PR #491 for that.

File: charts/hono/templates/NOTES.txt (hunk 16 above)
[the remark this answers is not in the document]
Comment (calohmn) [reply charts/hono/templates/NOTES.txt:17]: The files
`charts/hono/templates/example-data-grid/configmap.yaml`
and
`charts/hono/templates/hono-service-device-registry-base/hono-service-device-registry-post-install-configmap.yaml`
still need to be adapted (`{{- $_ := set $ "dot" $args.dot }}` still in there).

File: charts/hono/templates/_helpers.tpl [hunk 18]
```diff
@@ -14,21 +14,32 @@
...
+  {{- empty $nameOverride | ternary .dot.Chart.Name $nameOverride | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
 {{/*
 Create a default fully qualified app name.
 We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
 If release name contains chart name it will be used as a full name.
+
+The scope passed in is expected to be a dict with keys
+- (mandatory) "dot": the root (".") scope
```
Comment (calohmn) [inline charts/hono/templates/_helpers.tpl:34]: This should be removed again.

**Comment by calohmn** [comment]
@WatcherWhale Could you rebase and resolve the conflicts? PR #491 has been merged.

File: charts/hono/templates/_helpers.tpl [hunk 19]
```diff
@@ -14,9 +14,15 @@
 
 {{/*
 Expand the name of the chart.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+
+The scope passed in is expected to be a dict with keys
+- (mandatory) "dot": the root (".") scope
 */}}
 {{- define "hono.name" -}}
```
Comment (calohmn) [inline charts/hono/templates/_helpers.tpl:22]: I think it would be consistent to then also remove the "dot" dict usage in the "hono.name" and "hono.chart" named templates again.

File: charts/hono/templates/NOTES.txt (hunk 16 above)
[the remark this answers is not in the document]
Comment (calohmn) [reply charts/hono/templates/NOTES.txt:17]: In these 2 files, can you also restore usage of `.` instead of `$` in the `{{- tpl ( .Files.Get ...` line?

**Comment by WatcherWhale** [comment]
@sophokles73 are there any other changes, I still need to apply for the 2 remaining unresolved issues?

File: charts/hono/templates/_helpers.tpl (hunk 15 above)
[the remark this answers is not in the document]
Comment (sophokles73) [reply charts/hono/templates/_helpers.tpl]: I see

**Comment by sophokles73** [comment]
> are there any other changes, I still need to apply for the 2 remaining unresolved issues?

Well, as my comment, indicates, IMHO it would be good if you added a comment explaining what is actually being computed, as it is not really obvious to me ...

File: charts/hono/templates/_helpers.tpl [hunk 20]
```diff
@@ -14,34 +14,46 @@
...
 {{/*
 Expand the name of the chart.
+
+The scope passed in is expected to be a dict with keys
+- (mandatory) "dot": the root (".") scope
 */}}
 {{- define "hono.name" -}}
-  {{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+  {{- $nameOverride := .dot.Values.nameOverride -}}
+  {{- empty $nameOverride | ternary .dot.Chart.Name $nameOverride | trunc 63 | trimSuffix "-" -}}
```
[the remark this answers is not in the document]
Comment (WatcherWhale) [reply charts/hono/templates/_helpers.tpl]: I updated the comment and also restructured the code a bit to make it more legible and similar to what is used in `hono.fullname`

File: packages/cloud2edge/Chart.yaml [hunk 21]
```diff
@@ -11,8 +11,8 @@
 # SPDX-License-Identifier: EPL-2.0
 #
 apiVersion: v1
-version: 0.4.2
-appVersion: 0.4.2
+version: 0.4.1
+appVersion: 0.4.1
```
Comment (sophokles73) [inline packages/cloud2edge/Chart.yaml:15]: I do not think that the c2e package chart needs to be changed as part of this PR, or am I mistaken?

**Review by sophokles73 — CHANGES_REQUESTED** [review]

File: packages/cloud2edge/Chart.yaml (hunk 21 above)
Comment (calohmn) [reply packages/cloud2edge/Chart.yaml:15]: I guess that was a rebase-error - the c2e chart had been updated to 0.4.2 in the meantime.

Comment (WatcherWhale) [reply packages/cloud2edge/Chart.yaml]: It was indeed a rebase error I fixed it now

File: charts/hono/Chart.yaml [hunk 22]
```diff
@@ -15,7 +15,7 @@ name: hono
 description: |
   Eclipse Hono™ provides remote service interfaces for connecting large numbers of IoT devices to a back end and
   interacting with them in a uniform way regardless of the device communication protocol.
-version: 2.4.4
+version: 2.5.0
```
Comment (sophokles73) [inline charts/hono/Chart.yaml:18]: one final question:

Based on the information from the [When to increase a chart’s version number? section](https://www.eclipse.org/packages/contribute/) of the IoT Packages contribution guide, I would think that increasing the patch version should be sufficient for this kind of change, or am I missing something?

@calohmn WDYT?

Comment (calohmn) [reply charts/hono/Chart.yaml:18]: The changes in this PR change the name of the Hono resources if the chosen _release name_ doesn't contain "hono" (or the corresponding `nameOverride` for hono).
So, using the "eclipse-hono" release name (as documented in the Hono chart README), there would be no change to most of the names (only the Kafka related resources, I think).

But, when deploying Hono as part of the cloud2edge package, using `c2e` as release name, the names will change from
```
c2e-adapter-amqp
c2e-service-auth
c2e-ditto-gateway
c2e-ditto-nginx
```
to
```
c2e-hono-adapter-amqp
c2e-hono-service-auth
c2e-ditto-gateway
c2e-ditto-nginx
```
So, this is an incompatible change, requiring adaptations to the cloud2edge chart. That's why we have restricted the Hono chart version in the cloud2edge dependencies to `~2.4.0`.

So, we have to (at least) increase the Hono chart minor version here. (See also discussion in https://github.com/eclipse/packages/pull/416#discussion_r1001436531.)

**Review by sophokles73 — APPROVED** [review]
LGTM

**Review by calohmn — APPROVED** [review]
@WatcherWhale Thanks for bringing this to completion!
